pym/bob/state.py

```diff
@@ -608,6 +625,12 @@ def resetWorkspaceState(self, path, dirState):
             else:
                 self.__dirStates[path] = dirState
             needSave = True
+        if self.__layerStates.get(path) != dirState:
+            if dirState is None:
+                del self.__layerStates[path]
+            else:
+                self.__layerStates[path] = dirState
+            needSave = True
```

Comment:
Is this necessary? I think `resetWorkspaceState` should never be called for a layers directory, shouldn't it?